### PR TITLE
Moved `GetPodPriority` into a common library kubernetes#78220

### LINKED ISSUE #78220: Move `GetPodPriority` into a common library
**What would you like to be added**:

The `GetPodPriority` function should be moved out of `pkg/scheduler` and into a common library where components can use it without importing scheduler code.

**Why is this needed**:

In #76990 we removed a dependency between kubelet and scheduler by copying a small helper function. We would prefer to have only one copy of this function, but it would need a home outside of `pkg/scheduler` and `pkg/kubelet`

This package seems like a reasonable place: `k8s.io/kubernetes/pkg/apis/scheduling`

```go
func getPodPriority(pod *v1.Pod) int32 {
	if pod.Spec.Priority != nil {
		return *pod.Spec.Priority
	}
	// When priority of a running pod is nil, it means it was created at a time
	// that there was no global default priority class and the priority class
	// name of the pod was empty. So, we resolve to the static default priority.
	return scheduling.DefaultPriorityWhenNoDefaultClassExists
}
```

/sig scheduling
/help
/good-first-issue

### PR DESCRIPTION
**Pull request by dpandhi-git**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Removed dependency to load scheduler code for function GetPodPriority

**Which issue(s) this PR fixes**:
Fixes #78220 

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/scheduling/helpers_test.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ import (
 	"testing"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/api/core/v1"
```
Comment (ahg-g) [inline pkg/apis/scheduling/helpers_test.go]: This import is usually not named, just leave it as "k8s.io/api/core/v1" and use v1 in the code below as the package name.

**Review by ahg-g — COMMENTED** [review]
Thanks for doing this, note that in https://github.com/kubernetes/kubernetes/pull/76990/files#r286539095  a duplicate of getPodPriority is being introduced, if that PR gets merged first, then you will have to remove that redundant function and use the one you are adding here.

File: pkg/apis/scheduling/helpers_test.go (hunk 1 above)
Comment (dpandhi-git) [reply pkg/apis/scheduling/helpers_test.go]: Thanks @ahg-g I will update the code and remove the name for corev1 "k8s.io/api/core/v1" . 
How do I know if PR https://github.com/kubernetes/kubernetes/pull/76990/files#r286539095  gets merged first do I need to keep track of it?

Comment (ahg-g) [reply pkg/apis/scheduling/helpers_test.go]: yeah, I think you will just need to keep an eye on it. Most people are at kubecon hence the delay in reviews, hopefully some progress will be made this week.

**Comment by hprateek43** [comment]
Just went through your test results. The way you are arranging your build files is wrong. You probably need to refactor them as well. Also, make sure to run update bazel and verify go fmt from the ./hack folder. Do let me know if you need any help with that.

**Comment by misterikkit** [comment]
Hey, could we maybe move this helper into `/k8s.io/kubernetes/pkg/api/v1/pod/` ?
That package has other pod helper functions.
/remove-lifecycle stale

**Pull request closed without merging**